addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1457,19 +1461,32 @@ options.registry.topMenuColor = options.Class.extend({
     /**
      * @override
      */
-    selectStyle(previewMode, widgetValue, params) {
-        this._super(...arguments);
+    async selectStyle(previewMode, widgetValue, params) {
+        await this._super(...arguments);
         const className = widgetValue ? (params.colorPrefix + widgetValue) : '';
+        const color = _.intersection([className], this.$target[0].classList).join(' ');
+        const toggleOption = color && color.length ? 'header_color' : 'header_inline_color';
         this.trigger_up('action_demand', {
             actionName: 'toggle_page_option',
-            params: [{name: 'header_color', value: className}],
+            params: [{name: toggleOption, value: color  || widgetValue}],
         });
     },
 
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
 
+    /**
+     * @override
+     */
+    _computeWidgetState: function (methodName, params) {
+        if (!this.$el.hasClass('snippet-option-topMenuColor d-none')) {
+            const className = params.activeValue ? (params.colorPrefix + params.activeValue) : '';
+            this.$el.find('.selected').removeClass('selected');
+            this.$el.find('.o_we_color_preview').css('background-color','').removeClass(className);
+        }
+        return this._super(...arguments);
+    },
```

Comment:
Check how computeWidgetState works in other option. You are not supposed to alter the DOM in there, just returning a value.